src/Search/Search.Tests/Tests/SynonymMapTests.cs

```diff
@@ -0,0 +1,323 @@
+                searchClient.SynonymMaps.Create(synonymMap2);
+
+                SynonymMapListResult listResponse = searchClient.SynonymMaps.List();
+                Assert.Equal(2, listResponse.SynonymMaps.Count);
+
+                IEnumerable<string> synonymMapNames = listResponse.SynonymMaps.Select(i => i.Name);
+                Assert.Contains(synonymMap1.Name, synonymMapNames);
+                Assert.Contains(synonymMap2.Name, synonymMapNames);
+            });
+        }
+
+        [Fact]
+        public void ExistsReturnsTrueForExistingSynonymMap()
+        {
+            Run(() =>
+            {
+                SearchServiceClient client = Data.GetSearchServiceClient();
+                SynonymMap synonymMap = CreateTestSynonymMap();
+                client.SynonymMaps.Create(synonymMap);
+
+                Assert.True(client.SynonymMaps.Exists(synonymMap.Name));
+            });
+        }
+
+        [Fact]
+        public void ExistsReturnsFalseForNonExistingSynonymMap()
+        {
+            Run(() =>
+            {
+                SearchServiceClient client = Data.GetSearchServiceClient();
+                Assert.False(client.SynonymMaps.Exists("nonexistent"));
+            });
+        }
+
+        private void CreateAndValidateSynonymMap(SearchServiceClient searchClient, SynonymMap expectedSynonymMap)
+        {
+            SynonymMap actualSynonymMap = searchClient.SynonymMaps.Create(expectedSynonymMap);
+
+            try
+            {
+                AssertSynonymMapsEqual(expectedSynonymMap, actualSynonymMap);
+            }
+            finally
+            {
+                searchClient.SynonymMaps.Delete(expectedSynonymMap.Name);
+            }
+        }
+
+        private void CreateAndGetSynonymMap(SearchServiceClient searchClient, SynonymMap expectedSynonymMap)
+        {
+            searchClient.SynonymMaps.Create(expectedSynonymMap);
+
+            try
+            {
+                SynonymMap actualSynonymMap = searchClient.SynonymMaps.Get(expectedSynonymMap.Name);
+                AssertSynonymMapsEqual(expectedSynonymMap, actualSynonymMap);
+            }
+            finally
+            {
+                searchClient.SynonymMaps.Delete(expectedSynonymMap.Name);
+            }
+        }
+
+        private static void AssertSynonymMapsEqual(SynonymMap expected, SynonymMap actual)
+        {
+            Assert.Equal(expected, actual, new ModelComparer<SynonymMap>());
+        }
+
+        private static SynonymMap CreateTestSynonymMap()
+        {
+            return new SynonymMap(name: SearchTestUtilities.GenerateName(),
+                format: "solr",
+                synonyms: "word1,word2");
+        }
+
+        private static SynonymMap MutateSynonymMap(SynonymMap synonymMap)
+        {
+            synonymMap.Synonyms = "mutated1, mutated2";
+            return synonymMap;
+        }
+
+        private void TestAllSynonymMapFormats(Action<SearchServiceClient, SynonymMap> testMethod)
```

Comment:
This method doesn't seem useful given that we only support one synonym format right now. Maybe it's best to inline it for now.